lib/pipeline.js

```diff
@@ -1172,34 +1172,58 @@ class PipelineModel extends BaseModel {
 
         // Calculate event duration by using max endTime - min startTime of builds
         // that belong this event
-        const eventDuration = async (event) => {
-            const builds = await event.getBuilds();
+        const eventMetrics = async (event) => {
+            const buildMetrics = await event.getBuildMetrics();
 
             // Somehow this event doesn't have any builds
-            if (!builds || builds.length === 0) {
-                return 0;
+            if (!buildMetrics || buildMetrics.length === 0) {
+                return {
+                    duration: 0,
+                    queuedTime: 0,
+                    imagePullTime: 0
+                };
             }
 
-            let minStartTime = new Date(builds[0].startTime);
-            let maxEndTime = new Date(builds[0].endTime);
-
-            for (let i = 1; i < builds.length; i += 1) {
-                const s = new Date(builds[i].startTime);
-                const e = new Date(builds[i].endTime);
+            let minStartTime = new Date(buildMetrics[0].startTime);
+            let maxEndTime = new Date(buildMetrics[0].endTime);
+            let eventStatus = buildMetrics[0].status;
+            let totalQueuedTime = buildMetrics[0].queuedTime || 0;
+            let totalImagePullTime = buildMetrics[0].imagePullTime || 0;
+
+            for (let i = 1; i < buildMetrics.length; i += 1) {
+                const { startTime, endTime, queuedTime, imagePullTime, status } = buildMetrics[i];
+                const s = new Date(startTime);
+                const e = new Date(endTime);
+
+                // last build in the event
+                if (e > maxEndTime) {
+                    maxEndTime = e;
+                    eventStatus = status;
+                }
 
                 minStartTime = s < minStartTime ? s : minStartTime;
-                maxEndTime = e > maxEndTime ? e : maxEndTime;
+                if (queuedTime) {
+                    totalQueuedTime += queuedTime;
+                    totalImagePullTime += imagePullTime;
```

Comment:
better have a separate check for `imagePullTime `. Build can have `queueEnterTime`, `imagePullStartTime` but no `startTime` (failed when pulling image), in that case `imagePullTime` will be invalid